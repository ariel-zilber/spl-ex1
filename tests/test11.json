{
  "parties": [
    {
      "name": "The Baratheons",
      "mandates": 0,
      "join_policy": "M"
    },
    {
      "name": "The Targaryens",
      "mandates": 30,
      "join_policy": "M"
    },
    {
      "name": "The Starks",
      "mandates": 5,
      "join_policy": "M"
    },
    {
      "name": "The a",
      "mandates": 29,
      "join_policy": "M"
    },
    {
      "name": "The b",
      "mandates": 36,
      "join_policy": "M"
    },
    {
      "name": "The b",
      "mandates": 20,
      "join_policy": "M"
    }
  ],
  "graph": [
    [0, 1, 0, 0, 5, 3],
    [1, 0, 5, 0, 0, 3],
    [0, 5, 0, 1, 0, 3],
    [0, 0, 1, 0, 1, 1],
    [5, 0, 0, 1, 0, 3],
    [3, 3, 3, 1, 3, 0]
  ],
  "agents": [
    {
      "party_id": 0,
      "selection_policy": "E"
    },
    {
      "party_id": 1,
      "selection_policy": "E"
    }

  ]
}
  