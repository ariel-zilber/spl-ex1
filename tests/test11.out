[
    {
        "agents": [
            {
                "id": 0,
                "partyId": 0
            },
            {
                "id": 1,
                "partyId": 1
            }
        ],
        "coalitions": [
            {
                "mandates": 0,
                "parties": [
                    0
                ]
            },
            {
                "mandates": 30,
                "parties": [
                    1
                ]
            }
        ],
        "edges": [
            {
                "from": 0,
                "to": 1,
                "weight": 1
            },
            {
                "from": 0,
                "to": 4,
                "weight": 5
            },
            {
                "from": 0,
                "to": 5,
                "weight": 3
            },
            {
                "from": 1,
                "to": 0,
                "weight": 1
            },
            {
                "from": 1,
                "to": 2,
                "weight": 5
            },
            {
                "from": 1,
                "to": 5,
                "weight": 3
            },
            {
                "from": 2,
                "to": 1,
                "weight": 5
            },
            {
                "from": 2,
                "to": 3,
                "weight": 1
            },
            {
                "from": 2,
                "to": 5,
                "weight": 3
            },
            {
                "from": 3,
                "to": 2,
                "weight": 1
            },
            {
                "from": 3,
                "to": 4,
                "weight": 1
            },
            {
                "from": 3,
                "to": 5,
                "weight": 1
            },
            {
                "from": 4,
                "to": 0,
                "weight": 5
            },
            {
                "from": 4,
                "to": 3,
                "weight": 1
            },
            {
                "from": 4,
                "to": 5,
                "weight": 3
            },
            {
                "from": 5,
                "to": 0,
                "weight": 3
            },
            {
                "from": 5,
                "to": 1,
                "weight": 3
            },
            {
                "from": 5,
                "to": 2,
                "weight": 3
            },
            {
                "from": 5,
                "to": 3,
                "weight": 1
            },
            {
                "from": 5,
                "to": 4,
                "weight": 3
            }
        ],
        "parties": [
            {
                "id": 0,
                "mandates": 0,
                "name": "The Baratheons",
                "state": "Joined"
            },
            {
                "id": 1,
                "mandates": 30,
                "name": "The Targaryens",
                "state": "Joined"
            },
            {
                "id": 2,
                "mandates": 5,
                "name": "The Starks",
                "state": "Waiting"
            },
            {
                "id": 3,
                "mandates": 29,
                "name": "The a",
                "state": "Waiting"
            },
            {
                "id": 4,
                "mandates": 36,
                "name": "The b",
                "state": "Waiting"
            },
            {
                "id": 5,
                "mandates": 20,
                "name": "The b",
                "state": "Waiting"
            }
        ]
    },
    {
        "agents": [
            {
                "id": 0,
                "partyId": 0
            },
            {
                "id": 1,
                "partyId": 1
            }
        ],
        "coalitions": [
            {
                "mandates": 0,
                "parties": [
                    0
                ]
            },
            {
                "mandates": 30,
                "parties": [
                    1
                ]
            }
        ],
        "edges": [
            {
                "from": 0,
                "to": 1,
                "weight": 1
            },
            {
                "from": 0,
                "to": 4,
                "weight": 5
            },
            {
                "from": 0,
                "to": 5,
                "weight": 3
            },
            {
                "from": 1,
                "to": 0,
                "weight": 1
            },
            {
                "from": 1,
                "to": 2,
                "weight": 5
            },
            {
                "from": 1,
                "to": 5,
                "weight": 3
            },
            {
                "from": 2,
                "to": 1,
                "weight": 5
            },
            {
                "from": 2,
                "to": 3,
                "weight": 1
            },
            {
                "from": 2,
                "to": 5,
                "weight": 3
            },
            {
                "from": 3,
                "to": 2,
                "weight": 1
            },
            {
                "from": 3,
                "to": 4,
                "weight": 1
            },
            {
                "from": 3,
                "to": 5,
                "weight": 1
            },
            {
                "from": 4,
                "to": 0,
                "weight": 5
            },
            {
                "from": 4,
                "to": 3,
                "weight": 1
            },
            {
                "from": 4,
                "to": 5,
                "weight": 3
            },
            {
                "from": 5,
                "to": 0,
                "weight": 3
            },
            {
                "from": 5,
                "to": 1,
                "weight": 3
            },
            {
                "from": 5,
                "to": 2,
                "weight": 3
            },
            {
                "from": 5,
                "to": 3,
                "weight": 1
            },
            {
                "from": 5,
                "to": 4,
                "weight": 3
            }
        ],
        "parties": [
            {
                "id": 0,
                "mandates": 0,
                "name": "The Baratheons",
                "state": "Joined"
            },
            {
                "id": 1,
                "mandates": 30,
                "name": "The Targaryens",
                "state": "Joined"
            },
            {
                "id": 2,
                "mandates": 5,
                "name": "The Starks",
                "state": "CollectingOffers"
            },
            {
                "id": 3,
                "mandates": 29,
                "name": "The a",
                "state": "Waiting"
            },
            {
                "id": 4,
                "mandates": 36,
                "name": "The b",
                "state": "CollectingOffers"
            },
            {
                "id": 5,
                "mandates": 20,
                "name": "The b",
                "state": "Waiting"
            }
        ]
    },
    {
        "agents": [
            {
                "id": 0,
                "partyId": 0
            },
            {
                "id": 1,
                "partyId": 1
            }
        ],
        "coalitions": [
            {
                "mandates": 0,
                "parties": [
                    0
                ]
            },
            {
                "mandates": 30,
                "parties": [
                    1
                ]
            }
        ],
        "edges": [
            {
                "from": 0,
                "to": 1,
                "weight": 1
            },
            {
                "from": 0,
                "to": 4,
                "weight": 5
            },
            {
                "from": 0,
                "to": 5,
                "weight": 3
            },
            {
                "from": 1,
                "to": 0,
                "weight": 1
            },
            {
                "from": 1,
                "to": 2,
                "weight": 5
            },
            {
                "from": 1,
                "to": 5,
                "weight": 3
            },
            {
                "from": 2,
                "to": 1,
                "weight": 5
            },
            {
                "from": 2,
                "to": 3,
                "weight": 1
            },
            {
                "from": 2,
                "to": 5,
                "weight": 3
            },
            {
                "from": 3,
                "to": 2,
                "weight": 1
            },
            {
                "from": 3,
                "to": 4,
                "weight": 1
            },
            {
                "from": 3,
                "to": 5,
                "weight": 1
            },
            {
                "from": 4,
                "to": 0,
                "weight": 5
            },
            {
                "from": 4,
                "to": 3,
                "weight": 1
            },
            {
                "from": 4,
                "to": 5,
                "weight": 3
            },
            {
                "from": 5,
                "to": 0,
                "weight": 3
            },
            {
                "from": 5,
                "to": 1,
                "weight": 3
            },
            {
                "from": 5,
                "to": 2,
                "weight": 3
            },
            {
                "from": 5,
                "to": 3,
                "weight": 1
            },
            {
                "from": 5,
                "to": 4,
                "weight": 3
            }
        ],
        "parties": [
            {
                "id": 0,
                "mandates": 0,
                "name": "The Baratheons",
                "state": "Joined"
            },
            {
                "id": 1,
                "mandates": 30,
                "name": "The Targaryens",
                "state": "Joined"
            },
            {
                "id": 2,
                "mandates": 5,
                "name": "The Starks",
                "state": "CollectingOffers"
            },
            {
                "id": 3,
                "mandates": 29,
                "name": "The a",
                "state": "Waiting"
            },
            {
                "id": 4,
                "mandates": 36,
                "name": "The b",
                "state": "CollectingOffers"
            },
            {
                "id": 5,
                "mandates": 20,
                "name": "The b",
                "state": "CollectingOffers"
            }
        ]
    },
    {
        "agents": [
            {
                "id": 0,
                "partyId": 0
            },
            {
                "id": 1,
                "partyId": 1
            }
        ],
        "coalitions": [
            {
                "mandates": 0,
                "parties": [
                    0
                ]
            },
            {
                "mandates": 30,
                "parties": [
                    1
                ]
            }
        ],
        "edges": [
            {
                "from": 0,
                "to": 1,
                "weight": 1
            },
            {
                "from": 0,
                "to": 4,
                "weight": 5
            },
            {
                "from": 0,
                "to": 5,
                "weight": 3
            },
            {
                "from": 1,
                "to": 0,
                "weight": 1
            },
            {
                "from": 1,
                "to": 2,
                "weight": 5
            },
            {
                "from": 1,
                "to": 5,
                "weight": 3
            },
            {
                "from": 2,
                "to": 1,
                "weight": 5
            },
            {
                "from": 2,
                "to": 3,
                "weight": 1
            },
            {
                "from": 2,
                "to": 5,
                "weight": 3
            },
            {
                "from": 3,
                "to": 2,
                "weight": 1
            },
            {
                "from": 3,
                "to": 4,
                "weight": 1
            },
            {
                "from": 3,
                "to": 5,
                "weight": 1
            },
            {
                "from": 4,
                "to": 0,
                "weight": 5
            },
            {
                "from": 4,
                "to": 3,
                "weight": 1
            },
            {
                "from": 4,
                "to": 5,
                "weight": 3
            },
            {
                "from": 5,
                "to": 0,
                "weight": 3
            },
            {
                "from": 5,
                "to": 1,
                "weight": 3
            },
            {
                "from": 5,
                "to": 2,
                "weight": 3
            },
            {
                "from": 5,
                "to": 3,
                "weight": 1
            },
            {
                "from": 5,
                "to": 4,
                "weight": 3
            }
        ],
        "parties": [
            {
                "id": 0,
                "mandates": 0,
                "name": "The Baratheons",
                "state": "Joined"
            },
            {
                "id": 1,
                "mandates": 30,
                "name": "The Targaryens",
                "state": "Joined"
            },
            {
                "id": 2,
                "mandates": 5,
                "name": "The Starks",
                "state": "CollectingOffers"
            },
            {
                "id": 3,
                "mandates": 29,
                "name": "The a",
                "state": "Waiting"
            },
            {
                "id": 4,
                "mandates": 36,
                "name": "The b",
                "state": "CollectingOffers"
            },
            {
                "id": 5,
                "mandates": 20,
                "name": "The b",
                "state": "CollectingOffers"
            }
        ]
    },
    {
        "agents": [
            {
                "id": 0,
                "partyId": 0
            },
            {
                "id": 1,
                "partyId": 1
            },
            {
                "id": 2,
                "partyId": 2
            },
            {
                "id": 3,
                "partyId": 4
            }
        ],
        "coalitions": [
            {
                "mandates": 36,
                "parties": [
                    0,
                    4
                ]
            },
            {
                "mandates": 35,
                "parties": [
                    1,
                    2
                ]
            }
        ],
        "edges": [
            {
                "from": 0,
                "to": 1,
                "weight": 1
            },
            {
                "from": 0,
                "to": 4,
                "weight": 5
            },
            {
                "from": 0,
                "to": 5,
                "weight": 3
            },
            {
                "from": 1,
                "to": 0,
                "weight": 1
            },
            {
                "from": 1,
                "to": 2,
                "weight": 5
            },
            {
                "from": 1,
                "to": 5,
                "weight": 3
            },
            {
                "from": 2,
                "to": 1,
                "weight": 5
            },
            {
                "from": 2,
                "to": 3,
                "weight": 1
            },
            {
                "from": 2,
                "to": 5,
                "weight": 3
            },
            {
                "from": 3,
                "to": 2,
                "weight": 1
            },
            {
                "from": 3,
                "to": 4,
                "weight": 1
            },
            {
                "from": 3,
                "to": 5,
                "weight": 1
            },
            {
                "from": 4,
                "to": 0,
                "weight": 5
            },
            {
                "from": 4,
                "to": 3,
                "weight": 1
            },
            {
                "from": 4,
                "to": 5,
                "weight": 3
            },
            {
                "from": 5,
                "to": 0,
                "weight": 3
            },
            {
                "from": 5,
                "to": 1,
                "weight": 3
            },
            {
                "from": 5,
                "to": 2,
                "weight": 3
            },
            {
                "from": 5,
                "to": 3,
                "weight": 1
            },
            {
                "from": 5,
                "to": 4,
                "weight": 3
            }
        ],
        "parties": [
            {
                "id": 0,
                "mandates": 0,
                "name": "The Baratheons",
                "state": "Joined"
            },
            {
                "id": 1,
                "mandates": 30,
                "name": "The Targaryens",
                "state": "Joined"
            },
            {
                "id": 2,
                "mandates": 5,
                "name": "The Starks",
                "state": "Joined"
            },
            {
                "id": 3,
                "mandates": 29,
                "name": "The a",
                "state": "CollectingOffers"
            },
            {
                "id": 4,
                "mandates": 36,
                "name": "The b",
                "state": "Joined"
            },
            {
                "id": 5,
                "mandates": 20,
                "name": "The b",
                "state": "CollectingOffers"
            }
        ]
    },
    {
        "agents": [
            {
                "id": 0,
                "partyId": 0
            },
            {
                "id": 1,
                "partyId": 1
            },
            {
                "id": 2,
                "partyId": 2
            },
            {
                "id": 3,
                "partyId": 4
            },
            {
                "id": 4,
                "partyId": 5
            }
        ],
        "coalitions": [
            {
                "mandates": 56,
                "parties": [
                    0,
                    4,
                    5
                ]
            },
            {
                "mandates": 35,
                "parties": [
                    1,
                    2
                ]
            }
        ],
        "edges": [
            {
                "from": 0,
                "to": 1,
                "weight": 1
            },
            {
                "from": 0,
                "to": 4,
                "weight": 5
            },
            {
                "from": 0,
                "to": 5,
                "weight": 3
            },
            {
                "from": 1,
                "to": 0,
                "weight": 1
            },
            {
                "from": 1,
                "to": 2,
                "weight": 5
            },
            {
                "from": 1,
                "to": 5,
                "weight": 3
            },
            {
                "from": 2,
                "to": 1,
                "weight": 5
            },
            {
                "from": 2,
                "to": 3,
                "weight": 1
            },
            {
                "from": 2,
                "to": 5,
                "weight": 3
            },
            {
                "from": 3,
                "to": 2,
                "weight": 1
            },
            {
                "from": 3,
                "to": 4,
                "weight": 1
            },
            {
                "from": 3,
                "to": 5,
                "weight": 1
            },
            {
                "from": 4,
                "to": 0,
                "weight": 5
            },
            {
                "from": 4,
                "to": 3,
                "weight": 1
            },
            {
                "from": 4,
                "to": 5,
                "weight": 3
            },
            {
                "from": 5,
                "to": 0,
                "weight": 3
            },
            {
                "from": 5,
                "to": 1,
                "weight": 3
            },
            {
                "from": 5,
                "to": 2,
                "weight": 3
            },
            {
                "from": 5,
                "to": 3,
                "weight": 1
            },
            {
                "from": 5,
                "to": 4,
                "weight": 3
            }
        ],
        "parties": [
            {
                "id": 0,
                "mandates": 0,
                "name": "The Baratheons",
                "state": "Joined"
            },
            {
                "id": 1,
                "mandates": 30,
                "name": "The Targaryens",
                "state": "Joined"
            },
            {
                "id": 2,
                "mandates": 5,
                "name": "The Starks",
                "state": "Joined"
            },
            {
                "id": 3,
                "mandates": 29,
                "name": "The a",
                "state": "CollectingOffers"
            },
            {
                "id": 4,
                "mandates": 36,
                "name": "The b",
                "state": "Joined"
            },
            {
                "id": 5,
                "mandates": 20,
                "name": "The b",
                "state": "Joined"
            }
        ]
    },
    {
        "agents": [
            {
                "id": 0,
                "partyId": 0
            },
            {
                "id": 1,
                "partyId": 1
            },
            {
                "id": 2,
                "partyId": 2
            },
            {
                "id": 3,
                "partyId": 4
            },
            {
                "id": 4,
                "partyId": 5
            }
        ],
        "coalitions": [
            {
                "mandates": 56,
                "parties": [
                    0,
                    4,
                    5
                ]
            },
            {
                "mandates": 35,
                "parties": [
                    1,
                    2
                ]
            }
        ],
        "edges": [
            {
                "from": 0,
                "to": 1,
                "weight": 1
            },
            {
                "from": 0,
                "to": 4,
                "weight": 5
            },
            {
                "from": 0,
                "to": 5,
                "weight": 3
            },
            {
                "from": 1,
                "to": 0,
                "weight": 1
            },
            {
                "from": 1,
                "to": 2,
                "weight": 5
            },
            {
                "from": 1,
                "to": 5,
                "weight": 3
            },
            {
                "from": 2,
                "to": 1,
                "weight": 5
            },
            {
                "from": 2,
                "to": 3,
                "weight": 1
            },
            {
                "from": 2,
                "to": 5,
                "weight": 3
            },
            {
                "from": 3,
                "to": 2,
                "weight": 1
            },
            {
                "from": 3,
                "to": 4,
                "weight": 1
            },
            {
                "from": 3,
                "to": 5,
                "weight": 1
            },
            {
                "from": 4,
                "to": 0,
                "weight": 5
            },
            {
                "from": 4,
                "to": 3,
                "weight": 1
            },
            {
                "from": 4,
                "to": 5,
                "weight": 3
            },
            {
                "from": 5,
                "to": 0,
                "weight": 3
            },
            {
                "from": 5,
                "to": 1,
                "weight": 3
            },
            {
                "from": 5,
                "to": 2,
                "weight": 3
            },
            {
                "from": 5,
                "to": 3,
                "weight": 1
            },
            {
                "from": 5,
                "to": 4,
                "weight": 3
            }
        ],
        "parties": [
            {
                "id": 0,
                "mandates": 0,
                "name": "The Baratheons",
                "state": "Joined"
            },
            {
                "id": 1,
                "mandates": 30,
                "name": "The Targaryens",
                "state": "Joined"
            },
            {
                "id": 2,
                "mandates": 5,
                "name": "The Starks",
                "state": "Joined"
            },
            {
                "id": 3,
                "mandates": 29,
                "name": "The a",
                "state": "CollectingOffers"
            },
            {
                "id": 4,
                "mandates": 36,
                "name": "The b",
                "state": "Joined"
            },
            {
                "id": 5,
                "mandates": 20,
                "name": "The b",
                "state": "Joined"
            }
        ]
    },
    {
        "agents": [
            {
                "id": 0,
                "partyId": 0
            },
            {
                "id": 1,
                "partyId": 1
            },
            {
                "id": 2,
                "partyId": 2
            },
            {
                "id": 3,
                "partyId": 4
            },
            {
                "id": 4,
                "partyId": 5
            },
            {
                "id": 5,
                "partyId": 3
            }
        ],
        "coalitions": [
            {
                "mandates": 85,
                "parties": [
                    0,
                    4,
                    5,
                    3
                ]
            },
            {
                "mandates": 35,
                "parties": [
                    1,
                    2
                ]
            }
        ],
        "edges": [
            {
                "from": 0,
                "to": 1,
                "weight": 1
            },
            {
                "from": 0,
                "to": 4,
                "weight": 5
            },
            {
                "from": 0,
                "to": 5,
                "weight": 3
            },
            {
                "from": 1,
                "to": 0,
                "weight": 1
            },
            {
                "from": 1,
                "to": 2,
                "weight": 5
            },
            {
                "from": 1,
                "to": 5,
                "weight": 3
            },
            {
                "from": 2,
                "to": 1,
                "weight": 5
            },
            {
                "from": 2,
                "to": 3,
                "weight": 1
            },
            {
                "from": 2,
                "to": 5,
                "weight": 3
            },
            {
                "from": 3,
                "to": 2,
                "weight": 1
            },
            {
                "from": 3,
                "to": 4,
                "weight": 1
            },
            {
                "from": 3,
                "to": 5,
                "weight": 1
            },
            {
                "from": 4,
                "to": 0,
                "weight": 5
            },
            {
                "from": 4,
                "to": 3,
                "weight": 1
            },
            {
                "from": 4,
                "to": 5,
                "weight": 3
            },
            {
                "from": 5,
                "to": 0,
                "weight": 3
            },
            {
                "from": 5,
                "to": 1,
                "weight": 3
            },
            {
                "from": 5,
                "to": 2,
                "weight": 3
            },
            {
                "from": 5,
                "to": 3,
                "weight": 1
            },
            {
                "from": 5,
                "to": 4,
                "weight": 3
            }
        ],
        "parties": [
            {
                "id": 0,
                "mandates": 0,
                "name": "The Baratheons",
                "state": "Joined"
            },
            {
                "id": 1,
                "mandates": 30,
                "name": "The Targaryens",
                "state": "Joined"
            },
            {
                "id": 2,
                "mandates": 5,
                "name": "The Starks",
                "state": "Joined"
            },
            {
                "id": 3,
                "mandates": 29,
                "name": "The a",
                "state": "Joined"
            },
            {
                "id": 4,
                "mandates": 36,
                "name": "The b",
                "state": "Joined"
            },
            {
                "id": 5,
                "mandates": 20,
                "name": "The b",
                "state": "Joined"
            }
        ]
    }
]
